modules/social_features/social_profile/src/Plugin/views/access/SocialProfilePagesAccess.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Drupal\social_profile\Plugin\views\access;
+
+use Drupal\Core\Session\AccountInterface;
+use Drupal\views\Plugin\views\access\AccessPluginBase;
+use Symfony\Component\Routing\Route;
+
+/**
+ * Access plugin that provides access control to user information page.
+ *
+ * @ingroup views_access_plugins
+ *
+ * @ViewsAccess(
+ *   id = "profile_pages_access",
+ *   title = @Translation("Profile pages access"),
+ *   help = @Translation("Access to any profile page."),
+ * )
+ */
+class SocialProfilePagesAccess extends AccessPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function access(AccountInterface $account): bool {
+    return TRUE;
```

Comment:
This is potentially dangerous. I know we have this in a few other places but we shouldn't implement it this way.

In the current situation our route check is enough to deny access to the page. However, this plugin may be used in other ways (like a views generated block). In that case suddenly no access checking is happening anymore because this is hardcoded to `TRUE` and route checking doesn't apply.

We should implement the proper access logic (get the profile from context, then `$profile->access('view', $account);`. If that's not possible for some reason we should default this to `FALSE` with a comment of why we set this default value. Then worst case we have another "Access denied" bug where it shouldn't be denied, rather than having a security issue by allowing access where it shouldn't be allowed :)